flake.nix

```diff
@@ -0,0 +1,26 @@
+# https://github.com/numtide/devshell
```

Comment:
this file shouldn't have ended up in here, I made a seperate branch for it, but rebase go burr i guess.

I can't develop without it as it sets up node 16 (which is no longer maintained btw, so nixos considers it insecure).